platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/service/ReadableRosterStore.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.state.service;
+
+import com.hedera.hapi.node.state.roster.Roster;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * Provides read-only methods for interacting with Rosters.
+ */
+public interface ReadableRosterStore {
```

Comment:
The strategy you've taken here doesn't coincide with the PlatformState that @imalygin  and @tinker-michaelj  recently implemented.

For platform state, there is a `PlatformStateAccessor` interface, and then an implementation each for a readable and writable version.

Unless there is a reason to differ from that strategy, we should probably keep consistent.